### PR TITLE
This patch fixes STI class when methods are called on main Class.

### PR DESCRIPTION
**Pull request by n-rodriguez**
Example:
```
class Account < ApplicationRecord
  has_many :comments, as: :commentable
end

class Customer < Account
end

class Prospect < Account
end

class Comment < ActiveRecord::Base
  belongs_to :commentable, polymorphic: true
end
```
Now ```Customer.joins(:comments).count``` returns SQL query :
```
SELECT COUNT(*) FROM "accounts"
INNER JOIN "comments" ON "comments"."commentable_id" = "accounts"."id"
AND "comments"."commentable_type" = $1
WHERE "accounts"."type" IN ('Customer')  [["commentable_type", "Account"]]
```
With this patch it returns SQL query :
```
SELECT COUNT(*) FROM "accounts"
INNER JOIN "comments" ON "comments"."commentable_id" = "accounts"."id"
AND "comments"."commentable_type" = $1
WHERE "accounts"."type" IN ('Customer')  [["commentable_type", "Customer"]]
```

Edit : 
* this patch is already present in 4.1. It seems that it has been forgotten when patch for 4.2 was created.
* This PR comes with a test and it seems to work for older versions : https://travis-ci.org/appfolio/store_base_sti_class/builds/112963660, except for Rails 4.1.

### CODE REVIEW

**Comment by n-rodriguez** [comment]
Hi there! Any news?

File: lib/store_base_sti_class_for_4_1.rb [hunk 1]
```diff
@@ -308,7 +308,6 @@ def construct_join_attributes(*records)
             join_attributes
           end
         end
-
```
Comment (jonkessler) [inline lib/store_base_sti_class_for_4_1.rb]: Please undo these whitespace-only changes.

File: lib/store_base_sti_class_for_4_2.rb [hunk 2]
```diff
@@ -21,6 +21,7 @@ def creation_attributes
               # attributes[reflection.type] = owner.class.base_class.name
 
               attributes[reflection.type] = ActiveRecord::Base.store_base_sti_class ? owner.class.base_class.name : owner.class.name
+
```
Comment (jonkessler) [inline lib/store_base_sti_class_for_4_2.rb]: Please undo this.

**Comment by jonkessler** [comment]
I'm not gonna add a comment for each one, but could you undo the whitespace changes, please?

File: test/models.rb [hunk 3]
```diff
@@ -1,6 +1,20 @@
...
+
+class Customer < Account
+end
+
+class Prospect < Account
+end
+
+class Comment < ActiveRecord::Base
+  belongs_to :commentable, polymorphic: true
+end
```
Comment (jonkessler) [inline test/models.rb]: Can we use the existing test models rather than adding new ones?

**Comment by n-rodriguez** [comment]
What's the point of restoring white spaces?

**Comment by jonkessler** [comment]
Happy to merge when those comments are addressed and the build is green. Thanks for your contribution!

**Comment by n-rodriguez** [comment]
Gosh! If you don't want my patch I can remove it. Can you only accept to do your part of the job?

**Comment by jonkessler** [comment]
It's pretty standard practice to not accept whitespace changes in external pulls. I'm not sure what you mean by "Can you only accept to do your part of the job?" This is an open source project, and I'm spending time out of my workday to help you get this merged. I don't appreciate the comment.

**Comment by n-rodriguez** [comment]
> I'm not sure what you mean by "Can you only accept to do your part of the job?"

If you really want white spaces in your code, you can restore them if you want.

I'm also maintaining open source projects (and not only one : https://github.com/jbox-web, https://github.com/n-rodriguez?tab=repositories), and what I've learned from this experience : don't ask too much to people who also spend time out of their workday (and sometimes more) to contribute.
Being gratefull is already sufficient.

**Comment by jonkessler** [comment]
We are happy to merge when tests pass and your whitespace changes are separated out/part of another PR. Thanks for your contribution!

**Pull request closed without merging**